armi/reactor/blocks.py

```diff
@@ -1944,40 +1940,27 @@ def getPinCoordinates(self):
         Notes
         -----
         This assumes hexagonal pin lattice and needs to be upgraded once more generic geometry
-        options are needed.
-
-        A block with fully-defined pins could just use their individual spatialLocators in a
-        block-level 2-D grid. However most cases do not have this to minimize overhead and maximize
-        speed. Thus we want to just come up with a uniform mesh of pins if they're not explicitly
-        placed in the grid.
-
+        options are needed. Only works if pins have clad.
         """
-        return self._getPinCoordinatesHex()
-
-    def _getPinCoordinatesHex(self):
-        coordinates = []
-        numPins = self.getNumPins()
-        numPinRings = hexagon.numRingsToHoldNumCells(numPins)
-        pinPitch = self.getPinPitch()
-        if pinPitch is None:
-            return []
-        # pin lattice is rotated 30 degrees from assembly lattice
-        grid = grids.HexGrid.fromPitch(pinPitch, numPinRings, self, pointedEndUp=True)
-        for ring in range(numPinRings):
-            for pos in range(grid.getPositionsInRing(ring + 1)):
-                i, j = grid.getIndicesFromRingAndPos(ring + 1, pos + 1)
-                xyz = grid[i, j, 0].getLocalCoordinates()
-                coordinates.append(xyz)
-        return coordinates
+        coords = []
+        for clad in self.getChildrenWithFlags(Flags.CLAD):
+            if isinstance(clad.spatialLocator, grids.MultiIndexLocation):
+                coords.extend(
+                    [locator.getLocalCoordinates() for locator in clad.spatialLocator]
+                )
+            else:
+                coords.append(locator.getLocalCoordinates())
```

Comment:
Does this line work? I don't think the local variable `locator` is ever defined?

Reply:
no this line does not work.